common/src/main/java/io/homeassistant/companion/android/common/data/websocket/impl/WebSocketRepositoryImpl.kt

```diff
@@ -129,36 +144,39 @@ class WebSocketRepositoryImpl @Inject constructor(
 
     @ExperimentalCoroutinesApi
     override suspend fun getStateChanges(): Flow<StateChangedEvent> {
-        stateChangedMutex.withLock {
-            if (stateChangedFlow == null) {
+        try {
+            stateChangedMutex.withLock {
+                if (stateChangedFlow == null) {
 
-                val response = sendMessage(
-                    mapOf(
-                        "type" to "subscribe_events",
-                        "event_type" to "state_changed"
+                    val response = sendMessage(
+                        mapOf(
+                            "type" to "subscribe_events",
+                            "event_type" to "state_changed"
+                        )
                     )
-                )
 
-                stateChangedFlow = callbackFlow {
-                    producerScope = this
-                    awaitClose {
-                        Log.d(TAG, "Unsubscribing from state_changes")
-                        ioScope.launch {
-                            sendMessage(
-                                mapOf(
-                                    "type" to "unsubscribe_events",
-                                    "subscription" to response.id
+                    stateChangedFlow = callbackFlow {
+                        producerScope = this
+                        awaitClose {
+                            Log.d(TAG, "Unsubscribing from state_changes")
+                            ioScope.launch {
+                                sendMessage(
+                                    mapOf(
+                                        "type" to "unsubscribe_events",
+                                        "subscription" to response.id
+                                    )
                                 )
-                            )
+                            }
+                            producerScope = null
+                            stateChangedFlow = null
                         }
-                        producerScope = null
-                        stateChangedFlow = null
-                    }
-                }.shareIn(ioScope, SharingStarted.WhileSubscribed())
+                    }.shareIn(ioScope, SharingStarted.WhileSubscribed())
+                }
             }
-
-            return stateChangedFlow!!
+        } catch (e: Exception) {
+            Log.e(TAG, "Unable to get flow of entities", e)
         }
+        return stateChangedFlow!!
```

Comment:
This will more than likely throw another exception, we should return an `emptyFlow()`

Reply:
I was worried about that, done